### ISSUE TITLE
Why use test_loader instead of validation_loader?

### ISSUE THREAD

**Issue by missflash** [thuml/Anomaly-Transformer#50]
I think you should use `validation_loader` instead of `test_loader` in the following `self.vali` function.

https://github.com/thuml/Anomaly-Transformer/blob/e11c317974b729d330b8f84de0619902c3b5cee9/solver.py#L196

**Comment by umivic** [comment]
I agree with you. The test dataset should never be used as a validation set during the training process. This practice can lead to data leakage and result in overly optimistic performance evaluations. Any reported test performance metrics would be artificially high and not reflective of the model’s actual ability to generalize to unseen data.

**Comment by a1473838623** [comment]
Same question.
